### ISSUE TITLE
Unexpected Alerts will hang tests indefinitely

### ISSUE THREAD

**Issue by adrian-evo** [MarketSquare/robotframework-browser#674]
Hello,

I am currently investigating whether moving my project from Selenium to Browser library brings benefits, and first impression after few weeks of testing is that it is worth it. I have many test cases written for some web applications (Vaadin stateful framework) that could runs unattended for few hours. With selenium I had to use lower selenium speed (between 0.5 and 2 seconds), otherwise I would get invalid element state or stale element exceptions and so on. When using Browser library such issues don't exist so the run is a lot faster and more stable.

But..., currently I am facing a kind of a showstopper that would prevent me to move forward. Sometimes the applications could show an unexpected alert and this will hang the tests execution indefinitely. This means, if I start the testing and come back after few hours to check the result, or try to run the suite overnight, I could have an unhandled alert on the screen instead of having the tests run finished and reports generated, and this is unacceptable.

This does not happen with Selenium, and if the alert is unhandled there is an immediate error UnexpectedAlertPresentException, the test is failing and robot framework is moving naturally to the next test case from the suite.

I tried also to use Robot Framework's `Test Timeout`, but although the timeout is exceeded, the test is never finished.

**To Reproduce**
If needed, generate index.html with an alert and upload somewhere, e.g. with the following buttons:

```
<!doctype html>
<html>
  <head>
    <title>Title</title>
  </head>
  <body>
    <p>Content</p>
	<button type="button" onclick="javascript:alert('your msg')";>First click</button>
	<button type="button">Second click</button>
  </body>
</html>
```

Execute the following RF code:
```
    New Browser    chromium     headless=false
    New Context
    New Page       http://site/index.html
    Click  text=First click
    Click  text=Second click
```

**Expected behavior**
In the case of unhandled alert, I would expect the test is simply failing with an exception rather than being stuck indefinitely.

Alternatively, `Test Timeout` settings from Robot Framework should end the test case also in this situation.

Note:
When running such test directly with node and playwright, an unhandled alert can end the test with timeout exception if on the command line the flag `--unhandled-rejections=strict` is used.

**Comment by mkorpela** [comment]
Thank you for a very good problem description!

**Comment by mkorpela** [comment]
```
================= Original suppressed error =================
page.click: Timeout 10000ms exceeded.
=========================== logs ===========================
waiting for selector "text=First click"
  selector resolved to visible <button ;="" type="button" onclick="javascript:alert('yo…>First click</button>
attempting click action
  waiting for element to be visible, enabled and not moving
  element is visible, enabled and does not move
  scrolling into view if needed
  done scrolling
  checking that element receives pointer events at (43.77,60.48)
  element does receive pointer events
  performing click action
============================================================
Note: use DEBUG=pw:api environment variable and rerun to capture Playwright logs.
    at Connection.sendMessageToServer (/Users/mkorpela/workspace/rfbrowser/node_modules/playwright/lib/client/connection.js:69:15)
    at Proxy.<anonymous> (/Users/mkorpela/workspace/rfbrowser/node_modules/playwright/lib/client/channelOwner.js:44:61)
    at /Users/mkorpela/workspace/rfbrowser/node_modules/playwright/lib/client/frame.js:257:40
    at Frame._wrapApiCall (/Users/mkorpela/workspace/rfbrowser/node_modules/playwright/lib/client/channelOwner.js:72:34)
    at Frame.click (/Users/mkorpela/workspace/rfbrowser/node_modules/playwright/lib/client/frame.js:256:21)
    at /Users/mkorpela/workspace/rfbrowser/node_modules/playwright/lib/client/page.js:406:60
    at Page._attributeToPage (/Users/mkorpela/workspace/rfbrowser/node_modules/playwright/lib/client/page.js:205:20)
    at Page.click (/Users/mkorpela/workspace/rfbrowser/node_modules/playwright/lib/client/page.js:406:21)
    at Object.<anonymous> (/Users/mkorpela/workspace/rfbrowser/Browser/wrapper/index.js:70097:45)
    at step (/Users/mkorpela/workspace/rfbrowser/Browser/wrapper/index.js:69988:23) {
  name: 'TimeoutError'
}
```

**Comment by mkorpela** [comment]
Seems that the page click itself timeouts, also taking screen shot timeouts.

**Comment by mkorpela** [comment]
=> it gets stuck to the auto closing.

Setting `Library  Browser  auto_closing_level=MANUAL` stops the test

**Comment by mkorpela** [comment]
Seems that the problem is in our close_page.
Seems also that the page/context/browser switcher there does double work.

**Comment by mkorpela** [comment]
This has the bug: https://github.com/MarketSquare/robotframework-browser/blob/7e98fd54b41f221304bbdab7a8821b38e18ec56a/Browser/keywords/playwright_state.py#L207

**Comment by mkorpela** [comment]
Title does not timeout after the failed/timeouted click. Reproduced with pure playwright.

```
const {chromium} = require('playwright');

(async () => {
    const browser = await chromium.launch({ headless: false });
    const context = await browser.newContext();
    const page = await context.newPage();
    await page.goto('http://localhost:8000/');
    console.log("CLICK");
    try {
        await page.click('text=First click');
    } catch (e) {
        console.log("timeout! CLICK - done -> getting title");
        try {
            console.log(await page.title());
        } catch (e) {
            console.log("exception");
        }
        console.log("Title done")
    }
    await browser.close();
  }
)();
```

**Comment by mkorpela** [comment]
Fixed in #690
